astropy/table/column.py

```diff
@@ -1230,7 +1237,9 @@ def __setitem__(self, index, value):
 
         # update indices
         self.info.adjust_indices(index, value, len(self))
-        ma.MaskedArray.__setitem__(self, index, value)
+        with warnings.catch_warnings():
+            warnings.simplefilter('ignore', MaskedArrayFutureWarning)
```

Comment:
Theoretically, definitely possible. I can put `if ... else ...` here instead but not sure how much of a performance hit that would be, but maybe I'm pre-maturely attempting to optimize this.